Default to 755 permissions since checks later on test for non-group-writability.

Sage trac ticket: http://trac.sagemath.org/ticket/17875 
Upstream: https://bitbucket.org/pypa/setuptools/issue/254/the-default-python-eggs-mode-doesnt

diff -ur old/pkg_resources/__init__.py src/pkg_resources/__init__.py
--- old/pkg_resources/__init__.py	2015-02-16 14:57:32.000000000 +0100
+++ src/pkg_resources/__init__.py	2015-03-04 22:36:05.661039379 +0100
@@ -2955,7 +2955,7 @@
         os.makedirs(dirname)
 
 
-def _bypass_ensure_directory(path, mode=0o777):
+def _bypass_ensure_directory(path, mode=0o755):
     """Sandbox-bypassing version of ensure_directory()"""
     if not WRITE_SUPPORT:
         raise IOError('"os.mkdir" not supported on this platform.')

